backend/common/src/main/java/ai/verta/modeldb/common/subtypes/CommonArtifactHandler.java

```diff
@@ -122,15 +119,22 @@ public void logArtifacts(Handle handle, T entityId, List<Artifact> artifacts, bo
       validateAndThrowErrorAlreadyExistsArtifacts(entityId, artifacts, handle);
     }
 
-    for (final var artifact : artifacts) {
+    for (var artifact : artifacts) {
+      var path = artifact.getPath();
       var uploadCompleted = !artifactStoreConfig.getArtifactStoreType().equals(CommonConstants.S3);
       if (artifact.getUploadCompleted()) {
         uploadCompleted = true;
+      } else {
```

Comment:
I'm confused why we don't overwrite the path in this case. Shouldn't we overwrite the path _always_ since it's untrusted?

Reply:
Simplify logic and implemented it for nfs as well as per our discussion